docs/manual/usage/dotfiles.md

```diff
@@ -0,0 +1,36 @@
+# Keeping your \~ safe from harm {#sec-usage-dotfiles}
+
+To configure programs and services Home Manager must write various
+things to your home directory. To prevent overwriting any existing files
+when switching to a new generation, Home Manager will attempt to detect
+collisions between existing files and generated files. If any such
+collision is detected the activation will terminate before changing
+anything on your computer.
+
+For example, suppose you have a wonderful, painstakingly created
+`~/.config/git/config` and add
+
+``` nix
+{
+  # …
+
+  programs.git = {
+    enable = true;
+    userName = "Jane Doe";
+    userEmail = "jane.doe@example.org";
+  };
+
+  # …
+}
```

Comment:
Hmm, doesn't get rendered as Nix code for me. The classes in Firefox are `programlisting hljs language-bash`.

Had a quick look at the [generator code](https://github.com/NixOS/nixpkgs/blob/dc55b556b31a24f2a3e4c5d1c831b4c52f523746/pkgs/tools/nix/nixos-render-docs/src/nixos_render_docs/html.py#L99-L101) and it seems it ignores the block language?